dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ExecutorServiceImpl.java

```diff
@@ -262,12 +262,15 @@ private synchronized void createExecutor() {
         // ... and then make sure maxThreads is not smaller than coreThreads ...
         maxThreads = Math.max(coreThreads, maxThreads);
 
-        BlockingQueue<Runnable> workQueue = new BoundedBuffer<Runnable>(java.lang.Runnable.class, 1000, 1000);
+        BlockingQueue<Runnable> workQueue = new ConcurrentPriorityBlockingQueue<Runnable>();
 
         RejectedExecutionHandler rejectedExecutionHandler = new ExpandPolicy(workQueue, this);
 
         if (threadPool != null) {
-            ((BoundedBuffer<Runnable>) threadPool.getQueue()).removeFromAvailableProcessors();
+            BlockingQueue<Runnable> queue = threadPool.getQueue();
+            if (queue instanceof BoundedBuffer) {
+                ((BoundedBuffer<Runnable>) queue).removeFromAvailableProcessors();
+            }
```

Comment:
Is there any need to handle this situation for the case when `queue` is not an instance of `BoundedBuffer` in some way ?